TypeCobol/AntlrGrammar/CobolCodeElements.g4

```diff
@@ -8300,6 +8301,13 @@ correlation_clause: SQL_AS? (correlation_name=UserDefinedWord) new_column_names?
 new_column_names: LeftParenthesisSeparator new_column_name (SQL_CommaSeparator new_column_name)* RightParenthesisSeparator;
 new_column_name: UserDefinedWord;
 
+sqlRetain: ({ string.Equals(CurrentToken.Text, "RETAIN", System.StringComparison.OrdinalIgnoreCase) }? KeywordRETAIN=UserDefinedWord);
+sqlCursors: ({ string.Equals(CurrentToken.Text, "CURSORS", System.StringComparison.OrdinalIgnoreCase) }? KeywordCURSORS=UserDefinedWord);
+sqlLocks: ({ string.Equals(CurrentToken.Text, "LOCKS", System.StringComparison.OrdinalIgnoreCase) }? KeywordLOCKS=UserDefinedWord);
+onRollbackRetainCursorsClause: SQL_ON SQL_ROLLBACK sqlRetain sqlCursors;
+onRollbackRetainLocksClause: SQL_ON SQL_ROLLBACK sqlRetain sqlLocks;
+savepointStatement : SQL_SAVEPOINT (savepoint_name=UserDefinedWord) SQL_UNIQUE? SQL_ON SQL_ROLLBACK sqlRetain ((sqlCursors onRollbackRetainLocksClause?) | (sqlLocks onRollbackRetainCursorsClause));
```

Comment:
it should work but it's complicated, why not use: 

`savepointStatement: SQL_SAVEPOINT (savepoint_name=UserDefinedWord) SQL_UNIQUE? ((onRollbackRetainCursorsClause onRollbackRetainLocksClause?) | (onRollbackRetainLocksClause onRollbackRetainCursorsClause));`

?

This should also simplify the builder code.

Reply:
It doesn't work and what i proposed in the conception also doesn't work  that's why i changed .